### DISCUSSION TITLE
AbandonedRealms tt++ personal setup

### DISCUSSION THREAD

**Discussion by AbandonedRealms-tintin** [scandum/tintin#36] [Show and tell]
https://github.com/animalprimate/Abandonedrealms-tintin

Please check it out.  Please comment.  Thank you.

screenshot:
https://a.fsdn.com/con/app/proj/artt/screenshots/Screenshot%20at%202021-03-04%2017-11-10.png/max/max/1

**Comment by scandum** [comment]
Looks nice. One issue you'll run into for future releases, variables will need to start with a letter, so you either have to change `$05` to something else like `$log05` or use `${05}`

Most tmux features can be scripted now, including `#screen inputregion -2 1 -2 -1`, in case you want things to work in wintin++ as well.

**Comment by AbandonedRealms-tintin** [comment]
Thank you.

**Comment by AbandonedRealms-tintin** [comment]
Hi Scandum,  I've got a little bug here that keeps crawling back.

by the way I renamed my variables to log05, etc for future compatibility as
you suggested.

*** stack smashing detected ***: terminated
DEBUG_STACK[000] = poll_sessions(void)
DEBUG_STACK[001] = read_buffer_mud(0x5562428ac970)
DEBUG_STACK[002] = translate_telopts(0x5562428ac970,0x7ffc6ed0bdc0,2551)

I had to reduce amount of variables being set by my MSDP event to make this
go away.

Do you know more about this type of error as it relates to tintin++ and how
I can prevent?

**Reply by scandum** [reply]
I assume it's caused by a stack buffer overflow. If you can pin point which part of your script is causing the crash I can look into it.

I'm slowly working on overhauling memory management, so it should get fixed eventually.

**Comment by AbandonedRealms-tintin** [comment]
Thank you for your reply sir, I'll keep working on it!

**Comment by AbandonedRealms-tintin** [comment]
Now compatible with windows 10 via WSL (Windows Subsystems for Linux) 

1) Use search assistant to open the 'For Developers' setting and enable 'Developer Mode'.  Accept it.

2) Ctrl+Esc & type 'Turn Windows features on or off' select and turn on 'Windows Subsystem for Linux'

3) Open web browser and type 'https://aka.ms/wslstore' now select/install Ubuntu, set up and sign in.

4) Type 'sudo apt-get update' & 'sudo apt-get upgrade' & 'sudo apt-get install tintin++ imagemagick'

5) Download and run relevant wsltty installer from github: https://github.com/mintty/wsltty/releases

6) In WSL Terminal home directory, type 'git clone https://github.com/AbandonedRealms-tintin/ARTT.git'

7) In WSL Terminal type 'cd ~/ARTT' & 'chmod +x linux.sh' & './linux.sh' this will setup/launch game.
